mil_common/drivers/mil_passive_sonar/scripts/triggering

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+import rospy
+import numpy as np
+from rospy.numpy_msg import numpy_msg
+from std_msgs.msg import Header
+from std_srvs.srv import SetBool, SetBoolResponse
+from std_srvs.srv import Trigger, TriggerResponse
+from mil_passive_sonar.msg import HydrophoneSamples, Ping2, Ping
+from mil_passive_sonar import util
+from mil_passive_sonar.streamed_bandpass import StreamedBandpass
+from mil_ros_tools import Plotter, interweave
+
+from scipy.ndimage.filters import maximum_filter1d
+
+
+class HydrophoneTrigger:
+
+    def __init__(self):
+        '''ROS Node meant to trigger only when a ping happens in our target frequency range
+
+        Subscribes to raw hydrophone samples on `samples`
+
+        publishes samples(of gradient) from right around the triggering on `pings` when a ping is detected
+            and found to be in the target frequency range
+
+        publish a plot of the frequency response of the filter on `/filter_debug`  by service calling `/filter_debug_trigger`
+
+        '''
+
+        # Attriburtes about our general frequency range (all pinger live here)
+        #  Frequency range garunteed to be relatively quiet except for the pingers (in Hz)
+        self.general_lower = 15000 # lowest frequency pinger - 10 kHz
+        self.general_upper = 50000 # highest frequency pinger + 10 kHz
+
+        # Misc Attributes
+        # time (from start of samples)
+        self.time = 0.0
+        #  max convolution window, the amount of time that we assume should be quiet before a ping in sec
+        self.window_time = 0.1
+        #  prev data buffer
+        self.prev_data = None
+
+        # Ros stuff
+        rospy.init_node('hydrophone_trigger')
+        # if running with old bags use:
+        self.sub = rospy.Subscriber('samples', numpy_msg(Ping), self.hydrophones_cb)
+        self.sub = rospy.Subscriber('samples', numpy_msg(HydrophoneSamples), self.hydrophones_cb)
```

Comment:
?

Reply:
See most recent commit, I have changed how the support for Ping message type works